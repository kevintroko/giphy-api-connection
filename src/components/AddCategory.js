import React, { useState } from 'react';
import PropTypes from 'prop-types';

const AddCategory = ({setCategories}) => {
    const [value, setValue] = useState('');
    
    const handleInputChange = (e) => {
        console.log(e.target.value);
        setValue(e.target.value)
    }
    
    const handleSubmit = (e) => {
        e.preventDefault();
        
        if (value.trim().length > 2) {
            setCategories((categories) => [...categories, value]);
            setValue('');
        }
    }
    
    return (
    <form onSubmit={handleSubmit}>        
        <input 
            type="text"
            value={value}
            onChange={ handleInputChange }
        />
    </form>
)};

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}

export default AddCategory;
