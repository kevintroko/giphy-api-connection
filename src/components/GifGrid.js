import React from 'react'
const API_KEY = 'adWjSzzT0NRuUc0WQaiosvbKG9Xx13W4';

export const GifGrid = ({category}) => {
    const url = `https://api.giphy.com/v1/gifs/search?api_key=${API_KEY}&q=Liverpool&limit=5`;
    
    const getGifs = async () => {
        const resp = await fetch(url);
        console.log(resp);
        const {data} = await resp.json();
        const gifs = data.map(img => {
            return {
                id: img.id,
                title: img.title,
                url: img.images?.downsized_medium.url
            }
        })
        console.log(gifs);
    }
    
    getGifs();
    
    return (
        <div>
            <h3> {category} </h3>
        </div>
    )
}
