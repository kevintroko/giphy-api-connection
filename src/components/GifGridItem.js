import React from 'react'

export const GifGridItem = ({ id, title, url }) => {
    return (
        <div className="card">
            <img alt={title} src={url}/>
            <p>{title}</p>
        </div>
    )
}
