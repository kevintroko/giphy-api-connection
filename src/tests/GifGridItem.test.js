import React from 'react';
import { shallow } from "enzyme";
import { GifGridItem } from '../components/GifGridItem';

describe('Fist app', () => {
        const title = 'A title';
        const url = 'https://localhost/something.jpg';
    
        let wrapper;
    
        beforeEach(() => {    
            wrapper = shallow(<GifGridItem title={title} url={url}/>); 
        });
        
        test('should match snapshot', () => {  
            expect(wrapper).toMatchSnapshot();
        });    
});