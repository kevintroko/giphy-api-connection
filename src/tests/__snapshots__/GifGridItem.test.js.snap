// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fist app should match snapshot 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="A title"
    src="https://localhost/something.jpg"
  />
  <p>
    A title
  </p>
</div>
`;
